can/io/log.py

```diff
@@ -0,0 +1,95 @@
+from can.listener import Listener
+import datetime
+import time
+from can.message import Message
+
+CAN_MSG_EXT = 0x80000000
+CAN_ERR_FLAG = 0x20000000
+CAN_ERR_BUSERROR = 0x00000080
+CAN_ERR_DLC = 8
+
+class canutilsLogReader(object):
+    """
+    Iterator of CAN messages from a .log Logging File (candump -L).
+
+    Only CAN messages are supported (no Errorframes).
+    """
+
+    def __init__(self, filename):
+        self.fp = open(filename, "r")
+
+    def __iter__(self):
+        for line in self.fp:
+            temp = line.strip()
+            if len(temp) > 0:
+                (timestamp, bus, frame) = temp.split()
+                timestamp = float(timestamp[1:-1])
+                (canId, data) = frame.split("#")
+                if len(canId) > 3:
+                    isExtended = True
+                else:
+                    isExtended = False
+                canId = int(canId, 16)
+                if len(data) > 0 and (data[0] == "R" or data[0] == "r"):
+                    isRemoteFrame = True
+                    if len(data) > 1:
+                        dlc = int(data[1:])
+                    else:
+                        dlc = 0
+                else:
+                    isRemoteFrame = False
+
+                    dlc = len(data) / 2
+                    dataBin = bytearray()
+                    for i in range(0, int(dlc)):
```

Comment:
This might be simpler - one idea is to do your index math here using `range(0, 2 * dlc, 2)`.